src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTransformer.java

```diff
@@ -373,10 +375,20 @@ private StaticLangSettingsView generateSettingsClass(
     xsettingsClass.hasDefaultServiceAddress(interfaceConfig.hasDefaultServiceAddress());
     xsettingsClass.hasDefaultServiceScopes(interfaceConfig.hasDefaultServiceScopes());
     xsettingsClass.hasDefaultInstance(interfaceConfig.hasDefaultInstance());
+    xsettingsClass.generatorVersion(getGeneratorVersion());
 
     return xsettingsClass.build();
   }
 
+  private static String getGeneratorVersion() {
+    // TODO: Read version information from static file.
+    String version = JavaGapicSurfaceTransformer.class.getPackage().getImplementationVersion();
```

Comment:
Explanation: I think a TODO could make sense if we wouldn't do it for a while, or if it's a cleanup task that didn't affect the output, but I don't think either apply here.